modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ComponentMessageProcessor.java

```diff
@@ -403,16 +416,21 @@ private void onEvent(CoreEvent event, ExecutorCallback executorCallback) {
       final Map<String, Object> resolutionResult = sdkInternalContext.getResolutionResult(location, eventId);
 
       OperationExecutionFunction operationExecutionFunction = (parameters, operationEvent, callback) -> {
-        sdkInternalContext.setOperationExecutionParams(location, eventId, configuration, parameters, operationEvent, callback);
+        setOperationExecutionParams(location, event, configuration, parameters, operationEvent, callback, ctx);
 
         fluxSupplier.get().next(operationEvent);
       };
 
       if (location != null) {
         ((DefaultFlowCallStack) event.getFlowCallStack())
             .pushCurrentProcessorPath(resolvedProcessorRepresentation);
-        sdkInternalContext.getPolicyToApply(location, eventId)
-            .process(event, operationExecutionFunction, () -> resolutionResult, location, executorCallback);
+
+        ExecutorCallback effectiveCallback =
+            isTargetWithPolicies(event) ? getExecutionCallbackForPolicyAndOperationWithTarget(event, executorCallback)
```

Comment:
`isTargetWithPolicies(event)` is it really necessary to evaluate this method on every single invocation? Won't the value be constant? Cannot this be cached?

Reply:
No, which policies get applied to a particular operation depends on the incoming event